helix-term/src/handlers/completion.rs

```diff
@@ -291,6 +299,223 @@ fn request_completion(
     });
 }
 
+fn path_completion(
+    cursor: usize,
+    text: helix_core::Rope,
+    doc: &helix_view::Document,
+) -> Option<BoxFuture<'static, anyhow::Result<Vec<CompletionItem>>>> {
+    if !doc.supports_path_completion() {
+        return None;
+    }
+
+    use helix_lsp::util::range_to_lsp_range;
+    // TODO find a good regex for most use cases (especially Windows, which is not yet covered...)
+    // currently only one path match per line is possible in unix
+    static PATH_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"((?:~)?(?:\.{0,2}/)+.*)$").unwrap());
+
+    let cur_line = text.char_to_line(cursor);
+    let begin_line = text.line_to_char(cur_line);
+    let line_until_cursor = text
+        .slice(begin_line..cursor)
+        .as_str()
+        .unwrap_or_default()
+        .to_owned();
```

Comment:
With #9422 it should be possible to operate on the ropeslice directly now. The regex uses `$` though which doesn't work the same way with regex-cursor so it may not be possible to use it yet - regex-cursor and upstream regex might need the `\z` capture to detect the end of the slice

Reply:
Thanks for the suggestion. I'm using now the `regex-cursor`, as you say it doesn't support `$` the same way as `regex`, so I worked around that. But since the path detection is not perfect anyway (difficult, since everything apart of `\0` is allowed in unix paths). I think this is good enough for now.